test/e2e/test/elasticsearch/steps_mutation.go

```diff
@@ -97,6 +106,21 @@ func (b Builder) MutationTestSteps(k *test.K8sClient) test.StepList {
 		})
 }
 
+func IsOneNodeRollingUpgrade(b Builder) bool {
+	if b.MutatedFrom == nil {
+		return false
+	}
+	initial := b.MutatedFrom.Elasticsearch
+	mutated := b.Elasticsearch
+	// consider we're in the 1-node rolling upgrade scenario if we mutate
+	// from one node to one node with the same name
+	if len(initial.Spec.Nodes) == 1 && len(mutated.Spec.Nodes) == 1 &&
```

Comment:
Should we not rely on the data node count, in the same spirit that https://github.com/elastic/cloud-on-k8s/blob/855643a1cad50037b119f4339e1cdf320682829d/test/e2e/test/elasticsearch/settings.go#L13 ?

Reply:
Good point 👍